### ISSUE TITLE
SVG Embedding not working properly.

### ISSUE THREAD

**Issue by sgen** [agentk/fontfacegen#23]
See http://github.com/agentk/gulp-fontgen/issues/16

**Comment by sgen** [comment]
The issue persists.

**Comment by perrin4869** [comment]
If you look at the binary of `ttf2svg` [here](https://github.com/qdsang/ttf2svg/blob/master/bin/ttf2svg.js), it is outputting the file to the `pwd`, and outputs a log message, `'complete : '+fontOutputFilePath+'!'`, which is what is actually being saved as the svg file by `fontfacegen` (see the stdout redirection https://github.com/agentk/fontfacegen/blob/master/lib/ttf2svg.js#L15). We need to use the programmatic API presented in the README instead:

```js
var ttf2svg = require('ttf2svg')
  , fs = require('fs')
;

fs.readFile('./fontello.ttf', function (err, buffer) {
    if (!!err) throw err;

    var svgContent = ttf2svg(buffer);
    fs.writeFileSync('./fontello.svg', svgContent);

});
```

I will send a PR tomorrow with this fix.
The only tricky part would be to decide whether to use `ttf2svg` or `batik-ttf2svg`, and I don't have a mac where I can test it...

**Comment by sgen** [comment]
What OS are you using?

**Comment by perrin4869** [comment]
Slackware 14.1

**Comment by sgen** [comment]
I will test this on Ubuntu tonight and see if I can reproduce. This might be OS specific.

**Comment by perrin4869** [comment]
I doubt it, the source for `ttf2svg` is pretty clear, it works as expected. Check it out [here](https://github.com/qdsang/ttf2svg/blob/master/bin/ttf2svg.js)

Actually, the solution should be simple, if `isLinux` is true [here](https://github.com/agentk/fontfacegen/blob/master/lib/ttf2svg.js#L14), we use the programmatic API, i.e., `var ttf = require('ttf2svg')`, etc. Otherwise we use `batik-ttf2svg`

What problem does `batik-ttf2svg` solve again? Why can't we just use `ttf2svg` in all cases?

**Comment by sgen** [comment]
Im not actually sure batik-ttf2svg is necessary. I just didnt remove it in a prior commit because I couldnt test the end result on Mac OS.

**Comment by perrin4869** [comment]
I see! OK then I'll see if I can test `ttf2svg` on a friend's Mac tomorrow, and if it works, l think we can make it a dependency of `fontfacegen` and remove all references to `batik` here.

The only problem is that `ttf2svg` is no longer maintained anymore, so it might be a problem moving forward, but hopefully not.

**Comment by perrin4869** [comment]
Closed by #36
